experimental/dds/tree/src/interface-prototype/Examples.ts

```diff
@@ -0,0 +1,141 @@
+import { v4 } from 'uuid';
+// This file uses these as opaque id types:
+// the user of these APIs should not know or care if they are short IDs or not, other than that they must be converted to StableId if stored for use outside of the shared tree it was acquired from.
+// In practice, these would most likely be implemented as ShortId numbers.
+import { Definition, TraitLabel } from '../Identifiers';
+import { Side } from '../Snapshot';
+import { StableId, TreeNodeData } from './Anchors';
+import {
+	anchorDataFromNodeId,
+	Command,
+	CommandContext,
+	CommandId,
+	CommandRegistry,
+	PrefetchFilter,
+	root,
+	SharedTree,
+	TreeNodeViewReadonly,
+} from './Checkout';
+import { Place, TreeNode } from './MutableAnchors';
+
+//////////////// Command examples //////////////
+
+// Inserts a node with the specified Definition and identifier at the specified Place, and return it.
+export const insertExample = {
+	id: 'f73f004c-3b3e-42fe-b7c9-2e5e8793ca45' as CommandId,
+	run: (
+		context: CommandContext,
+		{ definition: def, identifier: id }: { definition: StableId; identifier: StableId },
+		{ place }: { place: Place }
+	): TreeNode => {
+		const definition = context.loadDefinition(def);
+		const identifier = context.loadNodeId(id);
+		const range = place.insert({ definition, identifier, traits: {} });
+		return range.start.adjacentNode(Side.After);
+	},
+};
+
+function newNodeId(): StableId {
+	return v4() as StableId;
+}
+
+export const doubleInsertExample = {
+	id: '08bac27d-632f-48bb-834a-90af8d67ca60' as CommandId,
+	run: (context: CommandContext, _: {}, { parent }: { parent: TreeNode }): TreeNode => {
+		const a = context.runCommand(
+			insertExample,
+			{ definition: context.stabilize(bar), identifier: newNodeId() },
```

Comment:
It's unclear to me why we need to stabilize bar if we're passing it to our own API. If it's a ShortId, shouldn't it still be the same across all contexts? (Yes, we can have temporary ShortIds, but I would have thought updates to those are a behind-the-scenes detail, and in any case a client can do anything about it while offline.)

Reply:
the extra options passed to Commands have to be able to be persisted and reapplied in another instance of the application when/of replaying the command. Since we have no reflection information for what the values in the options are, we have no way to convert what ever runtime Id representation we have into something more stable when persisting it (we can only tell that its a number, string, etc). Thus we have to use something stable directly.

I'm not happy with how this turned out: its pretty ugly. I don't see a better approach unless we just allow using the full persistable Ids more APIs (and add some more static types for them).

In realistic usage it would be less bad since you are likely to have the full Ids in scope from your schema objects, and also passing definitions to commands probably won't be common (I imagine anchors and various primitives will likely make up most of the arguments)

Reply:
I guess I still don't see why the ShortId wouldn't be stable, even if reapplied in another instance of the application replaying the command. If the command were to be applied to another *document*, that would be another matter. I suppose we *might* want to support that in some complex mutli-document simultaneous merge... but is that the scenario you had in mind or is there something simpler that I'm missing? One idea might be to put all tree-related parameters in the "anchors" block, though we might need a different name for it then.

Reply:
I believe you have some ShortID proposal somewhere thats relevant, and I assume based on your response provides stable ShortIds. Can you link it?